src/fieldImpl.cc

```diff
@@ -106,44 +94,48 @@ Field::addExt(lsst::afw::image::Wcs const& wcs_,
     /*
      * We're going to fake psfex's wcsstruct object.  We only need enough of it for field_locate
      */
-    PsfUnpack wcsUnpacked(wcs_);
-    struct wcsprm const* wcsPrm = wcsUnpacked.getWcsInfo();
     QMALLOC(impl->wcs[impl->next], wcsstruct, 1);
     wcsstruct *wcs = impl->wcs[impl->next];
     
-    wcs->naxis = wcsPrm->naxis;
+    wcs->naxis = 2;
     wcs->naxisn[0] = naxis1;
     wcs->naxisn[1] = naxis2;
-    
+
+    auto const crval = wcs_.getSkyOrigin();
+    // crpix using the FITS standard = pixel origin using the LSST standard + 1
+    auto const crpix = wcs_.getPixelOrigin() + afw::geom::Extent2D(1, 1);
+    auto const cdMatrix = wcs_.getCdMatrix();
+    std::string const cunit("DEG");
+    auto metadata = wcs_.getFitsMetadata();
     for (int i = 0; i != wcs->naxis; ++i) {
-        strncpy(wcs->ctype[i], wcsPrm->ctype[i], sizeof(wcs->ctype[i]) - 1);
-        strncpy(wcs->cunit[i], wcsPrm->cunit[i], sizeof(wcs->cunit[i]) - 1);
-        wcs->crval[i] = wcsPrm->crval[i];
-        
-        wcs->cdelt[i] = wcsPrm->cdelt[i];
-        wcs->crpix[i] = wcsPrm->crpix[i];
-        wcs->crder[i] = wcsPrm->crder[i];
-        wcs->csyer[i] = wcsPrm->csyer[i];
-        wcs->crval[i] = wcsPrm->crval[i];
+        auto ifits = i + 1;
+        auto ctype = metadata->getAsString("CTYPE" + std::to_string(ifits));
+        strncpy(wcs->ctype[i], ctype.c_str(), ctype.size() + 1);
+        strncpy(wcs->cunit[i], cunit.c_str(), cunit.size() + 1);
+        wcs->crpix[i] = crpix[i];
+        wcs->crval[i] = crval[i].asDegrees();
+        wcs->cdelt[i] = 1.0;  // scale is in the CD matrix (is this even needed?)
+        wcs->crder[i] = 0;
+        wcs->csyer[i] = 0;
     }
-    for (int i = 0; i != wcs->naxis*wcs->naxis; ++i) {
-        wcs->cd[i] = wcsPrm->cd[i];
+    for (int i = 0, k = 0; i < 2; ++i) {
+        for (int j = 0; j < 2; ++j, ++k) {
+            wcs->cd[k] = cdMatrix(i, j);
+        }
     }
-    wcs->longpole = wcsPrm->lonpole;
-    wcs->latpole = wcsPrm->latpole;
-    wcs->lat = wcsPrm->lat;
-    wcs->lng = wcsPrm->lng;
-    wcs->equinox = wcsPrm->equinox;
+    wcs->lng = 0;
```

Comment:
Again, I'm unsure about these parameters, but you are not setting some and fixing others.

Reply:
I think the only fields I am not setting are lonpole and latpole. The defaults should suffice and they are a nuisance to determine because they depend on the sky projection used. I suppose it might be worth seeing if they show up in the metadata and setting them if so.